scylla/src/transport/query_result.rs

```diff
@@ -311,20 +314,12 @@ mod tests {
     }
 
     fn make_not_rows_query_result() -> QueryResult {
-        let table_spec = TableSpec::owned("some_keyspace".to_string(), "some_table".to_string());
-
-        let column_spec = ColumnSpec {
-            table_spec,
-            name: "column0".to_string(),
-            typ: ColumnType::Int,
-        };
-
         QueryResult {
             rows: None,
             warnings: vec![],
             tracing_id: None,
             paging_state: None,
-            col_specs: vec![column_spec],
+            metadata: None,
             serialized_size: 0,
         }
     }
```

Comment:
This changes the semantics of this function. Doesn't it break tests?

Reply:
The test case was invalid. `rows` being `None` imply empty `col_specs`. Fortunately, no tests depended on them being nonempty.

Reply:
From what I can see it is used in methods that make the rows non empty: `make_rows_query_result`, `make_string_rows_query_result`. Now those methods produce QueryResult with rows, but with empty col_specs. Isn't that invalid?

Reply:
It indeed is invalid. I'll fix it in make_rows_query_result, make_string_rows_query_result.